File: Recursion_Array/allIndices.cpp
#include<iostream>
#include<vector>
using namespace std;

void solution(vector<int>& arr, int idx, int num)
{
    if(idx == arr.size())
    {
        return;
    }
    if(arr[idx] == num)
    {
        cout << idx << " ";
    }
    solution(arr, idx + 1, num);
}

int main()
{
    int n;
    cin >> n;
    vector<int> arr(n);
    for(int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    int num;
    cin >> num;
    solution(arr, 0, num);
    return 0;
}

File: Recursion_Array/allIndices.java
import java.util.Scanner;
public class allIndices
{
    public static Scanner scn = new Scanner(System.in);
    public static void solution(int[] arr, int idx, int num)
    {
        if(idx == arr.length)
        {
            return;
        }
        if(arr[idx] == num)
        {
            System.out.print(idx + " ");
        }
        solution(arr, idx + 1, num);
    }

    public static void main(String[] args)
    {
        int n = scn. nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; i++)
        {
            arr[i] = scn.nextInt();
        }
        int num = scn.nextInt();
        solution(arr, 0, num);

    }
}


